### PR TITLE
Remove the current company from the D&B Hierarchy page

### PR DESCRIPTION
**Pull request by rafenden**
## Description of change

Remove the current company from the D&B Hierarchy page to make the number of related companies consistent with the business details page.

## Test instructions

Go to the Global Ultimate page and check if this company is not listed with the rest of D&B related companies.
 
### Before

image

### After 

image

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"

- [ ] Has the branch been rebased to develop?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/companies/apps/dnb-hierarchy/controllers.js [hunk 1]
```diff
@@ -31,21 +31,35 @@ async function renderDnbHierarchy (req, res, next) {
...
+        token,
+        company.global_ultimate_duns_number,
+        config.paginationDefaultSize,
+        page
+      )
     )
 
     res.json({
-      count,
+      count: count,
```
Comment (paulgain) [inline src/apps/companies/apps/dnb-hierarchy/controllers.js]: `count,` will suffice.

File: test/functional/cypress/fixtures/index.js [hunk 2]
```diff
@@ -3,21 +3,22 @@ module.exports = {
...
-    lambdaPlc: require('./company/lambda-plc'),
-    marsExportsLtd: require('./company/mars-exports-ltd'),
-    minimallyMinimalLtd: require('./company/minimally-minimal-ltd'),
-    oneListCorp: require('./company/one-list-corp.json'),
-    someOtherCompany: require('./company/some-other-company.json'),
-    venusLtd: require('./company/venus-ltd.json'),
-    withContacts: require('./company/with-contacts.json'),
-    oneListTierDita: require('./company/one-list-tier-d-ita.json'),
-    addInteractionError: require('./company/add-interaction-error.json'),
+    archivedLtd: require('../../../../test/sandbox/fixtures/v4/company/company-archived.json'),
```
Comment (PippoRaimondi) [inline test/functional/cypress/fixtures/index.js]: this is pretty cool, one more good thing bringing sandbox into datahub repo

Comment (rafenden) [reply test/functional/cypress/fixtures/index.js]: Yep :)

File: test/functional/cypress/specs/companies/dnb-hierarchy-spec.js [hunk 3]
```diff
@@ -2,38 +2,92 @@ const { assertLocalHeader, assertBreadcrumbs } = require('../../support/assertio
...
+      'Home': urls.dashboard(),
+      'Companies': urls.companies.index(),
+      [company.name]: urls.companies.detail(company.id),
+      'Business details': urls.companies.businessDetails(company.id),
+      'Related companies': null,
     })
+  })
+
+  it('should render the list of companies', () => {
+    cy.get('#dnb-hierarchy')
```
Comment (PippoRaimondi) [inline test/functional/cypress/specs/companies/dnb-hierarchy-spec.js]: given this guy is an ID it should be placed in selectors, just to keep consistency...

File: test/functional/cypress/specs/companies/dnb-hierarchy-spec.js [hunk 4]
```diff
@@ -2,38 +2,92 @@ const { assertLocalHeader, assertBreadcrumbs } = require('../../support/assertio
...
+      .should('contain', '2 related company records')
+      .and('not.contain', company.name)
+      .get('h3')
+      .parent()
+      .should('have.length', companiesInHierarchy.length)
+      .each((x, i) => {
```
Comment (PippoRaimondi) [inline test/functional/cypress/specs/companies/dnb-hierarchy-spec.js]: giving x, i a more meaningful name could help make the test more readable

**Pull request merged**